protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -897,4 +900,35 @@ class _FieldSet {
 
     _oneofCases?.addAll(original._oneofCases!);
   }
+
+  _FieldSet deepCopy(GeneratedMessage message, {bool freeze = false}) {
+    final values = _makeValueList(_values.length);
+
+    for (var valueIdx = 0; valueIdx < _values.length; valueIdx += 1) {
+      final value = _values[valueIdx];
+      if (value is PbMap) {
+        values[valueIdx] = value.deepCopy(freeze: freeze);
+      } else if (value is PbList) {
+        values[valueIdx] = value.deepCopy(freeze: freeze);
+      } else if (value is GeneratedMessage) {
+        values[valueIdx] = value.deepCopy(freeze: freeze);
+      } else {
+        values[valueIdx] = value;
+      }
+    }
+
+    final copy = _FieldSet._(
+      message,
+      null, // TODO: event plugin
```

Comment:
Please leave a LDAP or an issue number in todos.

```suggestion
      null, // TODO(omersa): event plugin
```